### PR TITLE
Coupons: Add usage restrictions view

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #6491 

### Description

This PR updates the old CouponUsageDetails screen to a similar screen named Usage Restrictions.

Changes:
- Rename the classes relating to UsageDetails to UsageRestrictions.
- Update the UI of the old usage details view to match the new design.
- Add a new screen to add allowed emails. This screen is navigated from the Usage Restrictions screen.
- Remove the navigation link to usage details from the coupon details screen.

### Testing instructions

There's no entry point to this screen yet since the AddEditCoupon screen is still in progress. Please test the screen in the preview mode of Xcode.

In order to test the navigation to the allowed emails screen, feel free to wrap the body of `UsageRestrictions` within a `NavigationView`. I'm under the impression that the `AddEditCoupon` view will need to be wrapped inside a `NavigationView`, so there's no need for an additional navigation view in this view.

### Screenshots

This is a capture of the view with a navigation view wrapping the body:
https://user-images.githubusercontent.com/5533851/160802829-0e1272a5-9bba-4afc-9bb1-8b5412fd18af.mov

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Coupons/UsageDetails/CouponAllowedEmails.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+}
+
+private extension CouponAllowedEmails {
+    enum Constants {
+        static let margin: CGFloat = 16
+        static let topSpacing: CGFloat = 24
+    }
+
+    enum Localization {
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Coupons/UsageDetails/CouponAllowedEmails.swift:37]: nit: I am wondering, do we add the Localization strings on the view model as when working with UIKit, or in this case there is no view model?

**Comment by toupper** [comment]
Great job @itsmeichigo! 🚀  
One question, I see one in Xcode Preview for CouponRestrictions one row (Usage Limit per User) that I cannot locate in the designs, neither [here](https://woomobilep2.wordpress.com/2022/02/24/coupon-management-i2/) nor [here](https://github.com/woocommerce/woocommerce-ios/issues/6491). Is it an exceptional case, or are there a newer version of the designs?:

coupons-restrictions

**Review by toupper — APPROVED** [review]

File: WooCommerce/Classes/ViewRelated/Coupons/UsageDetails/CouponAllowedEmails.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Coupons/UsageDetails/CouponAllowedEmails.swift:37]: I believe we can add localized strings anywhere they're needed, either on the view model or the view itself or even both, as long as they're relevant. In this case, the strings are used for the UI so I'm adding the localization enum as the extension of the view.

**Comment by itsmeichigo** [comment]
> One question, I see one in Xcode Preview for CouponRestrictions one row (Usage Limit per User) that I cannot locate in the designs, neither [here](https://woomobilep2.wordpress.com/2022/02/24/coupon-management-i2/) nor https://github.com/woocommerce/woocommerce-ios/issues/6491. Is it an exceptional case, or are there a newer version of the designs?

Great question @toupper! This "Usage limit per user" field is a configuration in Core (you can find it in WP Admin > Marketing > Coupons > Add new coupon). I think it's somehow missed on the designs so I added it myself, to make sure that we have all the configurations like in the web app.

**Pull request merged**